pkg/subctl/cmd/gather/resource.go

```diff
@@ -26,27 +27,32 @@ import (
 )
 
 func ResourcesToYAMLFile(info *Info, ofType schema.GroupVersionResource, namespace string) error {
-	list, err := info.DynClient.Resource(ofType).Namespace(namespace).List(metav1.ListOptions{})
+	lists, err := info.DynClient.Resource(ofType).Namespace(namespace).List(metav1.ListOptions{})
 	if err != nil {
 		return errors.WithMessagef(err, "error listing %q", ofType.Resource)
 	}
-
-	path := filepath.Join(info.DirName, info.ClusterName+"-"+ofType.Resource+".yaml")
-	file, err := os.Create(path)
-	if err != nil {
-		return errors.WithMessagef(err, "error opening file %s", path)
+	if len(lists.Items) == 0 {
+		return errors.New(fmt.Sprintf("No config maps found in namespace %s", namespace))
 	}
 
-	defer file.Close()
+	for _, list := range lists.Items {
+		path := filepath.Join(info.DirName, info.ClusterName+"-"+ofType.Resource+"-"+list.GetName()+".yaml")
```

Comment:
```suggestion
		path := filepath.Join(info.DirName, info.ClusterName+"-" + namespace + "-"+ofType.Resource+"-"+list.GetName()+".yaml")
```

Reply:
To be complete, we can handle empty namespace:

```suggestion
                 nsPart := ""
                 if namespace != "" {
                     nsPath = "-" + namespace
                 }
                 
		path := filepath.Join(info.DirName, info.ClusterName+"-"+ofType.Resource+nsPart+"-"+list.GetName()+".yaml")
```